nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/service/loading.service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, delay, Observable } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class LoadingService {
+    status$: Observable<boolean>;
+
+    loading: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+    requests: Map<string, boolean> = new Map<string, boolean>();
+
+    constructor() {
+        this.status$ = this.loading.asObservable().pipe(delay(0));
```

Comment:
using `.pipe(delay(0))` is essentially equivalent to using a `setTimeout()`. I'm curious about the necessity of this in the current context, and I'm thinking that leveraging Angular Signals might be another viable option that could contribute positively to this situation.

What are your thoughts on this? I'm open to discussing and finding the most effective approach together.